### ISSUE TITLE
Broken promise

### ISSUE THREAD

**Issue by Wuntenn** [angulartics/angulartics#533]
I have a function (`function a`)that should normally return a promise. Under certain criteria, I throw an exception from this same function.

The invoking function (`function b`) handles the exception by catching it within the function-chain that called `function a` in the first place. The handler simply console.logs  the output - nothing fancy.

The problem: Angulartics keeps reporting this same exception as unhandled. I can see that exception track is registered in the config block and from what I've read takes over with exceptions handling to allow special types of errors like `GA`'s one to be handled.

It seems that this is not giving my handler a chance to handle the error and causes my code not to work.

Oddly, This has worked in many places within the same file, however for some reason I'm encountering issue with angulartics exceptionTrack in this instance.

I've have a couple other issues #529 recently and Angulartics [line 379](https://github.com/angulartics/angulartics/blob/master/src/angulartics.js#L379) seems to be at the root of most of them.

Is anyone able to advise me on ways to bypass this issue? Is there a way to disable exceptionTrack completely. I've looked at the docs which say to use: `$analyticsProvider.trackExceptions(true)` and the [source](https://github.com/angulartics/angulartics/blob/master/src/angulartics.js#L49). Both imply that whist exceptionTrack is off by default, it is still able to give me grief.

**Comment by Wuntenn** [comment]
I've since tested my theory by returning the `$delegate` unchanged. I get the same error, however this time it's reported by Angular's `hint.js`.

I'm baffled. I realise that this is no longer a concern to this project, however, just incase anyone can see anything glaringly wrong and feels like helping, my problem is the following:

`function a` should normally return a promise, however the exception is being thrown (which is good because I want to make sure it works). The problem is that the exception isn't being caught: 
```
function invokingFunction() {
  return somefunctionThatReturnsPromise()
    .then(innerPromiseReturningFunction)
    .then(anotherLikeThe first);

  function innerPromiseReturningFunction() {
    return promiseReturningFn()
      .catch(handleErr)
  } 
}

// problemFunction
function a() {
  if (rejection-criteria == true) throw new Error(' oh no rejection criteria');

  // Do lots of stuff...
  return promiseOfSomeKind;
}

function handleErr(err) {
  console.log('Oh no err: ', err);
}
```

**Comment by Wuntenn** [comment]
Fixed it

**Comment by timelf123** [comment]
What was the fix?

**Comment by Wuntenn** [comment]
Realising that I'm stupid (again!)

Once I realised that Angulartics was informing me the same error that I would've seen regardless of the exception handling, I started desperate process of elimination in the way that I handled the catch. 

I actually removed the throw that I was trying to catch above and it revealed another entirely new error that I'd had thrown elsewhere in the chain - That's the one it was complaining about, and the reason my catch here wasn't being triggered.

Because of where I caught the error the code lacked a result it was supposed to have and was able to proceed to trigger the second error. I fixed it by making sure that I handled that case better.

Thanks for helping. I'm going to have to play around with the catch to recreate this when I get a chance. Its a weird one!
